tests/test_routes.py

```diff
@@ -426,3 +414,58 @@ def test_get_item_order_not_found(self):
         """
         resp = self.app.get(f"{BASE_URL}/0/items/0")
         self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_no_item(self):
+        """It should not Update an item given wrong order id and non-existent order"""
+        order = self._create_orders(1)[0]
+        item = OrderItemFactory()
+
+        # add item
+        resp = self.app.post(
+            f"{BASE_URL}/{order.id}/items",
+            json=item.serialize(),
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        data = resp.get_json()
+        item.id = data["id"]
+        self.assertIsNotNone(data["id"])
+        self.assertEqual(data["order_id"], order.id)
+        self.assertEqual(data["product_id"], item.product_id)
+        self.assertEqual(data["quantity"], item.quantity)
+        self.assertEqual(data["price"], item.price)
+
+        # update non-existent item id
+        updated_item = OrderItemFactory()
+        updated_item.id = item.id + 123
+        resp = self.app.put(
+            f"{BASE_URL}/{order.id}/items/{updated_item.id}",
+            json=updated_item.serialize(),
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
```

Comment:
You should also check here what is not found, order or item. Check the message of the response like other test cases!